ops/model.py

```diff
@@ -3792,6 +3792,40 @@ def _ensure_loaded(self):
         assert self._notice.key == self.key
 
 
+class LazyCheckInfo:
+    """Provide lazily-loaded access to a Pebble check's info.
+
+    The attributes provided by this class are the same as those of
+    :class:`ops.pebble.CheckInfo`, however, the notice details are only fetched
+    from Pebble if necessary (and cached on the instance).
+    """
+
+    name: str
+    level: Optional[Union[pebble.CheckLevel, str]]
```

Comment:
Looking through the implementation, I think that this field is always set, either a raw string, or an enum value, where the latter may be `UNSET`, but the field is never `None.

Reply:
This is copied from `CheckInfo`, which should match. The docs there say:

> This can be :attr:`CheckLevel.ALIVE`, :attr:`CheckLevel.READY`, or None (level not set).

The `CheckInfo.__init__` accepts `None`, although I don't think anyone would normally be manually creating one.  `from_dict` will set it to `None` if there's no `level` key in the dictionary. In [the Pebble code](https://github.com/canonical/pebble/blob/ffb99d9d60faa946c1ccf6afd2d7c813b0c2030b/client/checks.go#L37) is seems like "unset" is the empty string (like the ops enum). It does seem like the [Pebble API](https://github.com/canonical/pebble/blob/ffb99d9d60faa946c1ccf6afd2d7c813b0c2030b/internals/daemon/api_checks.go#L56) will always return a "level" value.

@benhoyt do you know the story here? Was there an earlier version of Pebble that wouldn't necessarily have "level" in the check response? If so then probably we need to continue supporting that and it should stay as-is (although a comment like the one for change-id might be worthwhile). If "level" will have been there for all Pebble versions, then I think @dimaqq is correct that removing the ability for it to be `None` makes sense (I'd do that in a separate PR).

Reply:
No, the "level" field is optional in the check config (you don't have to set a level), and it's marked as "omitempty" for the JSON serialisation, meaning its omitted from the JSON if it's unset (empty string) in the plan.

Reply:
Ah, the `omitempty` is what I missed. Thanks! So this needs to stay as-is.